client/payment-details/summary/index.tsx

```diff
@@ -141,6 +150,174 @@ const composePaymentSummaryItems = ( {
 		},
 	].filter( isValueTruthy );
 
+const composeResolvedDisputeMessage = (
```

Comment:
I think the logic is complex because this is over generalised – a big case statement with a few branches. For example, when rendering each kind of date, there's a fallback in case it doesn't exist or is not relevant. 

Not a blocker - just a comment, if I was trying to refactor this I'd start with that. But again, I don't want to hold this PR up, we can iterate on things like that if they annoy us in future :)

Reply:
Now I understand better, it looks like (gut feeling alert!) there are 3-4 `CardFooter` components needed (one for each case), and a higher level component to pick which one (if any) to render. I'm not sure how that would work out in practice.

Reply:
I've refactored these to separate components (in a single file) to disambiguate the outcome of this logic in d532d23.

We now will have three distinct sub-components:
* `DisputeResolutionFooter`
	* `DisputeWonFooter`
	* `DisputeUnderReviewFooter`
	* `DisputeLostFooter`
* More will likely be introduced in #7191 
	* `InquiryUnderReview`
	* `InquiryClosed`

Let me know what you think!